queries/scala/variables.scm

```diff
@@ -0,0 +1,11 @@
+(val_definition
+  pattern: (identifier) @variable)
+
+(var_definition
+  pattern: (identifier) @variable)
+
+(val_declaration
+  name: (identifier) @variable)
+
+(var_declaration
+  name: (identifier) @variable)
```

Comment:
```suggestion
(identifier) @var
```

Otherwise you'd have to cover every single way a variable may show up. This is a bit over-inclusive since it includes fieldnames (like in `Main.arg2` in your example), but it can be fixed in util.lua adding this to `ignored_field_names`

```lua
  scala = {
    field_expression = { 'field' }
  }
```
(it doesn't let me make this sugestion as part of the review because there were no changes there)